service/src/main/java/org/ehrbase/plugin/AbstartPluginAspect.java

```diff
@@ -171,7 +121,7 @@ private <T> Chain<T> buildChain(Collection<T> extensionPointInterfaceList, T ide
    * @param <T> Class of the Extension-point
    * @return output after all Extension-points have been handelt
    */
-  private <X, R, T> R handleChain(
+  protected <X, R> R handleChain(
```

Comment:
The implementation seems a bit too convoluted and redundant code is required at the calling side.

::handleChain could be wrapped by a method with the following parameters:
- ProceedingJoinPoint 
- the method selector
- a function argsToInput
- a function inputToArgs

If getExtensionPointInterfaceList() is empty, pjp.proceed() is returned.
Else 
- the method chain is constructed
- for the inner call  the arguments for pjp::proceed are constructed via inputToArgs
- args is converted to input
- the method chain is called
- the result is returned